### ISSUE TITLE
DPUpload is taking much time and getting failed with - Authentication Failure

### ISSUE THREAD

**Issue by RahulRdec** [IBM-UrbanCode/Datapower-Configuration-Manager-UCD#3]
Hi All, I am new to the DCM and I am trying to upload the files from local to remote directory.

While upload, first of all it is taking time to check-access and the getting failed after uploading 3-4 files out of 23 and the error we are getting is like below. Also it is not consistent, some times it failing after 3 files, sometime after 6. 
BUILD FAILED
C:\Users\xxxxx\IBM\wmbt80\workspace\Generic.all.B2B\DataPower\Delivery\dcmDa
tapower\gsk.mwi.deploy.ant.xml:1201: The following error occurred while executi
ng this line:
C:\Software\DCM\tmp\upload.ant.xml:18: java.lang.RuntimeException: Failed uploa
ding file local:///xxxx/aah/xxxx.dpa : <dp:result
xmlns:dp="http://www.datapower.com/schemas/management">Authentication failure<
/dp:result>

Please help and advise.

**Comment by nhmathis** [comment]
Hey RadhulRdec,

We'll need the full output log to provide a complete analysis. Please upload the output logs as a file. Feel free to redact any sensitive information.

-Nick

**Comment by RahulRdec** [comment]
Hi Nhmathis,
Thanks for your response. Attached are the logs. Let me know if you want me to capture any specific information. 
[logs.txt](https://github.com/IBM-UrbanCode/Datapower-Configuration-Manager-UCD/files/484691/logs.txt)

**Comment by RahulRdec** [comment]
Hi Richgroot,

Thanks for your feedback but if you check the logs from start, check-access was passed and it failed after 4 file upload successfully. I am not sure if is because of session timeout in between upload.

**Comment by richgroot** [comment]
DCM doesn't have any concept of sessions, just individual SOMA requests such as set-file, so the timeout applies to each individual SOMA request rather than to a series of requests.

A general procedure for diagnosing any problem with DCM is to do the steps manually through the DP web browser.  In your case you might upload the fourth file in the web browser and see if there is a complaint.  Since it is an xsl file it probably isn't too large, so I'm curious what the problem might be.

You can add -Dcapturesoma=soma.txt to the ant command line and it will write the SOMA requests and responses to the file named "soma.txt" (choose any filename you like, of course).  Send that file to me (regroot at us dot ibm dot com) and I'll take a look at the actual requests and responses to see if they are schema-compliant.  Since those requests are the interface between DCM and DataPower that is the place to look to determine whether DCM or DataPower is the place to continue the investigation.

**Comment by RahulRdec** [comment]
Hi Richgroot, hope you have received the SOMA output file for this execution. Let me know if you need anything else to be captured.

**Comment by richgroot** [comment]
I replayed this set of SOMA requests on my test appliance and it worked fine, so this is a bit of a puzzle.  I'm currently running an IDG 7.2.0.8 appliance, which is a different model and a slightly more recent version than you have.  The problem you are having doesn't appear to be a DCM problem but is probably an issue either with the particular firmware on your appliance, or something about that particular appliance.  I believe that your best course of action is to open a DataPower PMR describing the problem and include the SOMA capture that you made.

**Comment by RahulRdec** [comment]
Thanks Richgroot.
One thing I observed today, while I am using <dcm:upload> to upload only files it is taking couple of minutes (11 minutes and still failed) and same time if I am trying to import my service build as zip file which includes all these files also, it took only 32 secs. 

I am not able to decide now if this is the issue with dcm or device.

**Comment by richgroot** [comment]
Rahul,

I'm pretty puzzled by what you are reporting, which is one reason I pointed
you toward Support with a PMR.  They may have seen this before and
recognize it.  When I tried the series of SOMA requests against a DP VM on
my laptop, they were close to instantaneous so the upload time that you are
seeing for each file is unusual, especially given that they are so small.
I'm will add timestamps to the capturesoma output in order to show the
times of each request and response but I don't know that will help you at
all.

Thanks,
Rich
